nixos/modules/services/hardware/udisks2.nix

```diff
@@ -21,23 +23,43 @@ with lib;
         '';
       };
 
+      mountAtMedia = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          By default, udisks2 mounts removable drives under the ACL
+          controlled directory /run/media/$USER/. If you wish to
+          mount to /media instead, enable this option.
+
+          This option will also clean /media at every boot.
+          Since /media, unlike /run, is not mounted by default as a
+          tmpfs, /media directory needs to be cleaned to get rid of
+          stale mountpoints.
+        '';
+      };
+
     };
 
   };
 
 
   ###### implementation
 
-  config = mkIf config.services.udisks2.enable {
+  config = mkIf cfg.enable {
 
     environment.systemPackages = [ pkgs.udisks2 ];
 
     services.dbus.packages = [ pkgs.udisks2 ];
 
-    systemd.tmpfiles.rules = [ "d /var/lib/udisks2 0755 root root -" ];
+    systemd.tmpfiles.rules = [ "d /var/lib/udisks2 0755 root root -" ]
+      ++ optional cfg.mountAtMedia "D! /media 0755 root root - ";
```

Comment:
I am somewhat queasy about this.

Also, is it just supposed to clean after mounts that were not unmounted at previous shutdown, or does udisks not clean after itself when unmounting?

Reply:
`udisksctl unmount` cleans itself, but calling poweroff/reboot etc. without calling `udisksctl` leads to stale folders.

Reply:
This is a good idea as `/run/media` is already a `tmpfs`, but `UDISKS_FILESYSTEM_SHARED` environment variable kinda hardcodes the shared directory as `/media`.

Reply:
I read some more documentation from upstream and I was mistaken in my assumption. Those `tmpfiles` rules assume mounting to `/run/media/$USER` which I'm guessing is not what you want...

What is your main motivation in this PR? If we better understand what your goal is maybe we can find the best solution easier.

My goal here is to keep us inline with upstream intentions and other distros.

Reply:
Ubuntu also encourages this here: https://sites.google.com/site/installationubuntu/home/ubuntu-20-04/change-mountpoint-of-usb-drives

When a mount point is created under `/run/media/$USER`, it is only accessible to that user. This is burden for a, lets say,  media center PC where multiple users would like to access the mount point.